src/Symfony/Component/BrowserKit/Client.php

```diff
@@ -141,9 +141,9 @@ public function setServerParameter($key, $value)
      * Gets single server parameter for specified key.
      *
      * @param string $key     A key of the parameter to get
-     * @param string $default A default value when key is undefined
+     * @param mixed  $default A default value when key is undefined
```

Comment:
do we want this? At first sight i dont see any test relying on it...

Reply:
on branch 4.4 at least:
1) Symfony\Component\BrowserKit\Tests\AbstractBrowserTest::testXmlHttpRequest
Failed asserting that '' is false.

Reply:
hm, i would patch 4.4 instead. It's a type violation IMHO.

Reply:
or a BC break :)
REQUEST_TIME_FLOAT isn't a string either so that forcing a string here wouldn't be good

Reply:
float values can be represented as string :) for now i'd rely on the fact we dont use strict types: https://3v4l.org/3lAC8

getting server parameters as strings seems really valueable IMHO.

Reply:
float to string to float isn't always accurate, we shouldn't take the risk

Reply:
fair enough.. maybe server params arent strings per se :sweat_smile: